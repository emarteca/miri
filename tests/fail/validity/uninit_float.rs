#![allow(deprecated)]
// This test is adapted from https://github.com/rust-lang/miri/issues/1340#issue-600900312.

fn main() {
    // Deliberately using `mem::uninitialized` to make sure that despite all the mitigations, we consider this UB.
    let _val: f32 = unsafe { std::mem::uninitialized() };
    //~^ ERROR: constructing invalid value at .value: encountered uninitialized bytes, but expected initialized bytes
}
