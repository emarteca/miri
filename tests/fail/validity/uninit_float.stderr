error: Undefined Behavior: constructing invalid value at .value: encountered uninitialized bytes, but expected initialized bytes
  --> $DIR/uninit_float.rs:LL:CC
   |
LL |     let _val: f32 = unsafe { std::mem::uninitialized() };
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^ constructing invalid value at .value: encountered uninitialized bytes, but expected initialized bytes
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: backtrace:
   = note: inside `main` at $DIR/uninit_float.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error

